datalad/support/gitrepo.py

```diff
@@ -1678,21 +1746,10 @@ def get_remote_branches(self):
         # TODO: treat entries like this: origin/HEAD -> origin/master'
         # currently this is done in collection
 
-        # For some reason, this is three times faster than the version below:
-        remote_branches = list()
-        for remote in self.repo.remotes:
-            try:
-                for ref in remote.refs:
-                    remote_branches.append(ref.name)
-            except AssertionError as e:
-                if str(e).endswith("did not have any references"):
-                    # this will happen with git annex special remotes
-                    pass
-                else:
-                    raise e
-        return remote_branches
-        # return [branch.strip() for branch in
-        #         self.repo.git.branch(r=True).splitlines()]
+        return [
+            b['refname'][13:]  # strip 'refs/remotes/'
+            for b in self.for_each_ref_(fields='refname', pattern='refs/remotes')
```

Comment:
I was going to remark that, if a "<remote>/HEAD" symbolic ref exists, this will return it, which we probably don't want.  But it turns out that the gitpy variant does the same thing.  OK.